kythe/proto/repo.proto

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package kythe.proto.repo;
+
+// Represents configuration data on how to extract a repo.
+message Config {
+  repeated BuildTarget target = 1;
+}
+
+// A single target in this repo to extract.
+message BuildTarget {
+  BuildSystem build_system = 1;
+  // The root of this extraction target, relative to the repo's root.
+  // The build config (e.g. pom.xml) should live in this directory.
+  string root = 2;
+
+  JavaOptions java_options = 3;
```

Comment:
We probably want the options per language to be highly extensible so maybe this should be a repeated Any field.

Reply:
offline: current rough plan is to go the other way.  Make things very coarse here, but then allow a given repo to fully specify a kytheextract.yaml or something if they need extremely custom behavior.

Anyways, some changes here.